aiomysql/sa/connection.py

```diff
@@ -103,7 +103,18 @@ def _base_params(self, query, dp, compiled, is_update):
                     "and execution with parameters"
                 )
         elif isinstance(query, ClauseElement):
-            compiled = query.compile(dialect=self._dialect)
+            if self._compiled_cache is not None:
+                key = query
+                compiled = self._compiled_cache.get(key)
+                if not compiled:
+                    compiled = query.compile(dialect=self._dialect)
+                    dp_checks = [dp and dp.keys() == compiled.params.keys()
```

Comment:
Perhaps a comment here what this list comprehension does, for a non sqlalchemy user, I have no clue what this does.

Reply:
Resolving this conversation due to the checks being removed :)